### PR TITLE
Add docstrings to lms_grades file

### LINKED ISSUE #128: Add docstrings and types to lms_grades.py
Add docstrings and types to all classes and functions defined in `src/illumidesk/handlers/lms_grades.py`.

### PR DESCRIPTION
**Pull request by jgwerner**
Closes #128

### CODE REVIEW

File: src/tests/illumidesk/apis/test_jupyterhub_api.py [hunk 1]
```diff
@@ -59,6 +60,27 @@ def test_initializer_sets_headers_to_make_requests(jupyterhub_api_environ):
...
+def test_request_builds_correct_url_variable(jupyterhub_api_environ):
+    """
+    Does the _request build the correct url based on the api url attribute and the
+    endpoint argument?
+    """
+    endpoint = 'users'
+    api_root_url = 'https://localhost/hub/api'
+    actual = f'{api_root_url}/{endpoint}'
+    expected = 'https://localhost/hub/api/users'
+    assert expected == actual
```
Comment (netoisc) [inline src/tests/illumidesk/apis/test_jupyterhub_api.py:81]: hm i think it's incomplete, right? why not use the wrapper?

File: src/illumidesk/apis/jupyterhub_api.py [hunk 2]
```diff
@@ -26,35 +29,35 @@ class JupyterHubAPI(LoggingConfigurable):
     def __init__(self):
         self.client = AsyncHTTPClient()
         self.token = os.environ.get('JUPYTERHUB_API_TOKEN')
-        if self.token is None:
+        if not self.token:
```
[the remark this answers is not in the document]
Comment (jgwerner) [reply src/illumidesk/apis/jupyterhub_api.py:32]: Yes agree, I'll add a check for both.

File: src/illumidesk/apis/jupyterhub_api.py [hunk 3]
```diff
@@ -26,35 +29,35 @@ class JupyterHubAPI(LoggingConfigurable):
...
         self.default_headers = {
             'Authorization': f'token {self.token}',
             'Content-Type': 'application/json',
         }
 
-    async def _request(self, endpoint, **kwargs):
+    async def _request(self, endpoint: str, **kwargs: Any) -> Awaitable['HTTPResponse']:
         """
-        Sends a request to the JupyterHub endpoint.
+        Wrapper for the AsyncHTTPClient.fetch method which adds additional log outputs.
```
[the remark this answers is not in the document]
Comment (jgwerner) [reply src/illumidesk/apis/jupyterhub_api.py:44]: Correct.

File: src/illumidesk/handlers/lms_grades.py [hunk 4]
```diff
@@ -19,6 +20,22 @@ def authenticator_class(self):
         return self.settings.get('authenticator_class', None)
 
     async def post(self, course_id: str, assignment_name: str) -> None:
+        """
+        Receives a request with the course name and the assignment name as path parameters
+        which then uses the LTIGradeSender.send_grades function to extract data from the
```
[the remark this answers is not in the document]
Comment (jgwerner) [reply src/illumidesk/handlers/lms_grades.py:25]: Got it, this was written pre grades send feature.

File: src/tests/illumidesk/apis/test_jupyterhub_api.py [hunk 5]
```diff
@@ -7,18 +7,20 @@
 from illumidesk.apis.jupyterhub_api import JupyterHubAPI
 
 
-def test_initializer_raises_error_without_JUPYTERHUB_API_TOKEN_env():
+def test_initializer_raises_error_with_jupyterhub_api_token_env_as_missing(monkeypatch, jupyterhub_api_environ):
     """
-    Does initializer raise an error when 'JUPYTERHUB_API_TOKEN' was not set?
+    Does initializer raise an error when 'JUPYTERHUB_API_TOKEN' is an empty string?
```
[the remark this answers is not in the document]
Comment (jgwerner) [reply src/tests/illumidesk/apis/test_jupyterhub_api.py:12]: In a separate test. Needless to say, a lot of our tests need parameterization.

File: src/tests/illumidesk/apis/test_jupyterhub_api.py (hunk 1 above)
Comment (jgwerner) [reply src/tests/illumidesk/apis/test_jupyterhub_api.py:81]: This is just a simple test to ensure the endpoint string is built correctly regardless of the client used to call this endpoint.

**Pull request merged**